### DISCUSSION TITLE
REM wont come online

### DISCUSSION THREAD

**Discussion by lockeandev** [tagyoureit/nodejs-poolController#987] [REM]
I had to turn power off to my Intellicenter OCP and when it came back online I made the mistake of trying to have it set its time from the internet, I think that allowed it to update the firmware. so I rolled it back to 1.064 but I'm till having problems with REM. probably not related to the OCP firmware.  Below is the output from pm2 monit logs for REM:

Any ideas on how to get REM back up and running?

![image](https://github.com/tagyoureit/nodejs-poolController/assets/11232490/0cf8bf93-211f-4144-a233-78e345c95615)

**Comment by lockeandev** [comment]
I cleaned the npm cache and updates all the apps (REM, dashPanel, njsPC) and cleaned the npm cache. I no longer get an error, but REM still does not ever come online. not I just see the following repeat over and over:
![image](https://github.com/tagyoureit/nodejs-poolController/assets/11232490/22cf5486-02a8-4ad7-9cad-c75e96232a19)

**Reply by tagyoureit** [reply]
From your first screenshot, the app is trying to write to a file that is open by another process.  Do you have REM starting up twice by any chance?  

You can run `ps -ef | grep node` and see all the node processes running on your system.

**Reply by lockeandev** [reply]
![image](https://github.com/tagyoureit/nodejs-poolController/assets/11232490/e8f8bedf-608e-4a4f-b506-7d8624b06603)

**Reply by lockeandev** [reply]
I deleted the directories for all three applications and reinstalled per https://github.com/tagyoureit/nodejs-poolController/wiki/Quick-Start-Guide and did not complet eht pm2 startup portion. 

If I start REM via npm start command, it works, but then if I reboot, and try a 2nd time  I get the EBUSY error.

**Reply by tagyoureit** [reply]
You have too many (5) dist/app.js commands running.  Did you previously have PM2 running the apps?  You can run `pm2 ls` to list or `pm2 stop all` to stop the processes.

**Reply by lockeandev** [reply]
Yes. PM2 was running fine for over two years. The power outage and un expected firmware upgrade to the intellicenter seems to have set off this most recent problem. 'pm2 ls' just shows the four expected apps. REM, njspc, dash panel, logrotate.

I can stop all and confirm they are stopped. And if I try to use npm start to start REM. I get the same error in the console....

**Reply by tagyoureit** [reply]
If you use `pm2 monit` (or `pm2 logs rem`) does it also show the same error?  The only explanations I can think of is that two processes are trying to write to the file; the disk is full (you'd be seeing lots of problems), or maybe that individual file became write protected or the user running the process doesn't have the right permissions?

**Reply by lockeandev** [reply]
yes, pm2 logs show the same error. I tried setting the file permissions for the whole REM folder so that anyone has full permissions.  No change. I tried another uninstall and reinstall. same thing REM works the first time, but if I pm2 stop all and pm2 start all, the next time it starts up, it throws the error. I have a 32 GB sd card, and 18 GB free. 

![image](https://github.com/tagyoureit/nodejs-poolController/assets/11232490/db301f6c-ccbb-47e7-b327-9659dca8f021)

**Reply by lockeandev** [reply]
Ok, so I think I figured it out. I think I had 2 gpio pins set to try to write out the same export? Not sure how that happened, but it must have been user error on my part. I restored my controller config from an older backup and it works fine now. I guess I must have mad a change and never rebooting the pi in 3 or 4 months. I just never noticed the error until I had a hard reboot due to a power outage this week.
